README.md

```diff
@@ -269,6 +269,16 @@ A `JsonApiClient::ResultSet` object should be paginatable with both `kaminari` a
 Person.where(name: 'Jeff').all
 ```
 
+## Headers
+
+You can specify the headers to send with all requests within a block with `with_headers`.
```

Comment:
Yeah, this will definitely not work in a multi-threaded environment. I would suggest either using a thread local variable for `@custom_headers` or passing it to some object instance that is thread safe (like a request object). The way this is implemented, I don't think there's an elegant way aside from the thread-local bit. Personally I think it might fit more elegantly into the syntax if it were just another method you called to modify the request: `Person.with_headers(authorization: "Token #{access_token}").where(name: "Jeff").all`